File: android/src/main/kotlin/com/shareinvest/flutter_kakao_maps/util/LogStreamHandler.kt
package com.shareinvest.flutter_kakao_maps.util

import io.flutter.plugin.common.EventChannel

class LogStreamHandler : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, sink: EventChannel.EventSink?) {
        eventSink = sink
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }

    fun sendMessage(message: String) {
        eventSink?.success(message)
    }
}

File: ios/flutter_kakao_maps/Sources/flutter_kakao_maps/util/LogStreamHandler.swift
import Flutter

class LogStreamHandler: NSObject, FlutterStreamHandler{
    var sink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
    
    func sendMessage(_ message: String) {
        guard let sink = sink else { return }
        
        sink(message)
    }
}
